resources/java_memory_assistant/bin/clean-up.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/sh -e
```

Comment:
Granted, shell is not a facility easy to test, and I am very open for proposal. I still think it is better to keep the upload of the heap-dump outside of the JVM, in particular if the upload should rely on 3rd party libraries to be bundled. But it would make perfect sense to use a better language to come up with those small utilities.

(Btw: would it be viable in terms of compliance and architecture to integrate Amazon's own SDKs for the upload to S3)

I think good options would be Ruby (platform independent and it can be tested more or less with the rest of the buildpack's code) or GoLang (and there I would set up a separate "satellite" project because of the different build & test pipeline). Ideas?